src/map-marker/location-puck.tsx

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright (c) Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+
+import * as React from 'react';
+import { useStyletron } from '../styles';
+import { getOverrides } from '../helpers/overrides';
+import {
+  LOCATION_PUCK_SIZES,
+  LOCATION_PUCK_TYPES,
+  EARNER_LOCATION_PUCK_CORE_SCALES,
+} from './constants';
+import type {
+  LocationPuckProps,
+  ConsumerLocationPuckProps,
+  EarnerLocationPuckProps,
+} from './types';
+import {
+  LocationPuckContainer,
+  StyledConsumerLocationPuckCore,
+  StyledLocationPuckApproximation,
+  StyledEarnerLocationPuckCore,
+} from './styled-components';
+
+const ConsumerLocationPuckHeading = ({ heading }) => {
+  const [css, theme] = useStyletron();
+  return (
+    <svg
+      className={css({
+        height: `${6}px`,
+        width: `${11}px`,
+        position: 'absolute',
+        color: theme.colors.contentAccent,
+        transform: `rotate(${heading}deg) translateY(-16px)`,
```

Comment:
`heading` is likely going to regularly change right? If so, let's move this css rule to `style` prop. Right now, styletron will create a new css class each time this value changes leading to a bunch of browser repaints from the new class added. 

```tsx
<svg
  style={{
    transform: `rotate(${heading}deg) translateY(-16px)`,
  }}
/>
```